common/src/main/java/org/apache/uniffle/common/ShuffleDataResult.java

```diff
@@ -34,12 +35,29 @@ public ShuffleDataResult(byte[] data) {
     this(data, Lists.newArrayList());
   }
 
-  public ShuffleDataResult(byte[] data, List<BufferSegment> bufferSegments) {
+  public ShuffleDataResult(List<BufferSegment> bufferSegments, ByteBuffer data) {
```

Comment:
> please swap the parameter order to `public ShuffleDataResult(ByteBuffer data, List<BufferSegment> bufferSegments) {`
> 
> it's aligned with previous constructor.

It will cause grammar error.

Reply:
My mistake. It is caused by error test code.